sources/gerritDB/index.js

```diff
@@ -164,15 +164,90 @@ var gerritDB = (function () {
+            + " where date_format(patch_set_approvals.granted, '%Y-%m-%d') >=" + "'" + filterDate + "'"
+            + " and patch_set_approvals.category_id = 'Code-Review'"
+            + " and changes.owner_account_id != accounts.account_id"
+            + " and full_name != 'Jenkins'";
+
+        doQuery(query, function (queryResult) {
+            callback(queryResult);
+        })
+    }
+
+    function getAverageMergeDurationByProject (filterOption, callback) {
+        var query = "SELECT Project, AVG(MergeProcessTime) AvgMergeDuration FROM ("
+            + " SELECT RealDiff,WorkingDayDiff,ChangeID, Project, CreatedTime, MergeTime, "
+            + " CASE WorkingDayDiff WHEN 0 THEN MergeTime ELSE DATE_SUB(MergeTime ,INTERVAL "
+            + " (((RealDiff-WorkingDayDiff)*24 +(WorkingDayDiff *14)))HOUR) END calculatedMergeTime,"
+            + " TIMESTAMPDIFF(SECOND,CreatedTime, CASE WorkingDayDiff WHEN 0 THEN MergeTime ELSE DATE_SUB(MergeTime, "
+            + " INTERVAL (((RealDiff-WorkingDayDiff)*24 + (WorkingDayDiff*14)))HOUR) END) / 60 MergeProcessTime"
+            + " FROM ("
+            + " SELECT DATEDIFF(MAX(Changes.last_updated_on),FirstUploadOfPatch.createdTime) RealDiff,"
+            + " dest_project_name Project,"
+            + " ABS(5 * (DATEDIFF(FirstUploadOfPatch.createdTime, MAX(Changes.last_updated_on)) DIV 7) + "
+            + " MID('0123444401233334012222340111123400012345001234550', 7 * WEEKDAY(FirstUploadOfPatch.createdTime) +"
+            + " WEEKDAY(MAX(Changes.last_updated_on)) + 1, 1))  WorkingDayDiff,"
+            + " FirstUploadOfPatch.change_id ChangeID, FirstUploadOfPatch.patch_set_id, MAX(Changes.last_updated_on) "
+            + " MergeTime,FirstUploadOfPatch.createdTime CreatedTime"
+            + " FROM gerrit.change_messages Message"
+            + " INNER JOIN gerrit.changes Changes ON Changes.change_id = Message.change_id"
+            + " INNER JOIN (SELECT patches.change_id,patch_set_id,uploader_account_id account_id, "
+            + " MIN(patches.created_on) AS createdTime FROM gerrit.patch_sets patches"
+            + " INNER JOIN gerrit.changes changes ON changes.change_id=patches.change_id AND changes.status='M'"
+            + " GROUP BY patches.change_id,patch_set_id,uploader_account_id"
+            + ") AS FirstUploadOfPatch ON"
+            + " FirstUploadOfPatch.change_id = Message.patchset_change_id AND"
+            + " FirstUploadOfPatch.patch_set_id = Message.patchset_patch_set_id AND FirstUploadOfPatch.patch_set_id =1"
+            + " WHERE Message.author_id <> 32 /* 32 is Jenkins account id. Should be changed according to your "
+            + "configurations*/ AND FirstUploadOfPatch.account_id <> Message.author_id"
+            + " GROUP BY FirstUploadOfPatch.change_id,FirstUploadOfPatch.patch_set_id"
+            + " ) AS MergeStatisticsView ) ProjectMerge"
+            + " WHERE Project NOT LIKE '%Onboarding%'"
+            + "GROUP BY Project;"
```

Comment:
can you please reform this sql for better readability? few notes:

* Spanning comments to multiple lines should be fixed (i.e.: line 206, 207)
* Inner blocks would be more visible if you indent them
* Also some comments about how the SQL really works would be nice since it's pretty hard to maintain this otherwise